### ISSUE TITLE
Switch to semver

### ISSUE THREAD

**Issue by freiheit** [EDCD/EDMarketConnector#523]
In particular, since moving from python 2 to python 3 will introduce major incompatibilities, make the python 3.7 release EDMC 4.0.0

https://semver.org/

**Comment by Athanasius** [comment]
Yup, I had bumping to 4.0.0.0 / 4.00 in mind, although not for this reason.

I'm convinced.

**Comment by Athanasius** [comment]
TODO List:

- [X] WinSparkle already operates from the full A.B.C.D version string.
- [ ] Update documentation so we're using `rel-A.B.C.D` as the release tag for github.

**Comment by Athanasius** [comment]
Need to check on the consequences of changing the release tags to `Release/A.B.C.D`.  A quick check showed github just handles the `/` without issues in the resulting paths.

So far I've not spotted anywhere that PyCharm treats **tags** as folder-able though.  It does do that for *branches*.  Thus I might stick with `rel-A.B.C.D`.

There's the possibility of then going back and changing *all* the old release tags to that new format for clarity, but that would mean also needing to change the tag associated with each release.  There's probably no actual gain to be had there due to what I said about PyCharm and tags above.

**Comment by Athanasius** [comment]
Just because PyCharm doesn't show `Release/A.B.C.D` **tags** in a folder-like manner doesn't mean other tools won't.

So, yes, move to that for tags as well as branches.

**Comment by Athanasius** [comment]
`Release/A.B.C.D` is now documented in [Contributing.md](https://github.com/EDCD/EDMarketConnector/blob/master/Contributing.md)

**Comment by freiheit** [comment]
From reading this, note that semver is MAJOR.MINOR.PATCH, so 4.0.0, not 4.0.0.0 or 4.00.

> Summary
> -------
> 
> Given a version number MAJOR.MINOR.PATCH, increment the:
> 
> 1. MAJOR version when you make incompatible API changes,
> 1. MINOR version when you add functionality in a backwards compatible
>    manner, and
> 1. PATCH version when you make backwards compatible bug fixes.
> 
> Additional labels for pre-release and build metadata are available as extensions
> to the MAJOR.MINOR.PATCH format.

[...]

> 1. A pre-release version MAY be denoted by appending a hyphen and a
> series of dot separated identifiers immediately following the patch
> version. Identifiers MUST comprise only ASCII alphanumerics and hyphens
> [0-9A-Za-z-]. Identifiers MUST NOT be empty. Numeric identifiers MUST
> NOT include leading zeroes. Pre-release versions have a lower
> precedence than the associated normal version. A pre-release version
> indicates that the version is unstable and might not satisfy the
> intended compatibility requirements as denoted by its associated
> normal version. Examples: 1.0.0-alpha, 1.0.0-alpha.1, 1.0.0-0.3.7,
> 1.0.0-x.7.z.92, 1.0.0-x-y-z.--.
> 
> 1. Build metadata MAY be denoted by appending a plus sign and a series of dot
> separated identifiers immediately following the patch or pre-release version.
> Identifiers MUST comprise only ASCII alphanumerics and hyphens [0-9A-Za-z-].
> Identifiers MUST NOT be empty. Build metadata MUST be ignored when determining
> version precedence. Thus two versions that differ only in the build metadata,
> have the same precedence. Examples: 1.0.0-alpha+001, 1.0.0+20130313144700,
> 1.0.0-beta+exp.sha.5114f85, 1.0.0+21AF26D3----117B344092BD.

**Comment by Athanasius** [comment]
I see this as a "fix the EDMC documentation" issue rather than a "Change EDMC to exactly use SEMVER" issue.

The whole way the WiX build and WinSparkle checking works is *know* to work with A.B.C.D and I won't be changing that just yet.

Although given that the 'D' makes no difference for at least WinSparkle I might be minded to change to just A.B.C at some point.  I also made an issue about supporting both `-<git hash>` and `+<fork>` in it without anything breaking.  The last I checked the build process in setup.py doesn't like anything extra being on the version.
